src/pages/Dashboard.tsx

```diff
@@ -122,6 +172,38 @@ const Dashboard = () => {
             <Button onClick={createNewInsight}>
                 New Insight
             </Button>
+            {/**TODO: Popup form for new group**/}
+            <div>
+                <form onSubmit={() => createNewGroup(newGroupNameInput)}>
+                    <label>
+                        New Group Name:
+                        <input
+                            type="text"
+                            value={newGroupNameInput}
+                            onChange={changeEvent => setNewGroupNameInput(changeEvent.target.value)}
+                            minLength={3}
+                            maxLength={25}
+                        />
+                    </label>
+                    <input type="submit" value="Submit" />
+                </form>
+            </div>
+            <div>
+                <span>Groups:</span>
+                <ul>
+                    {currentUserGroups.map(group => (
+                        <li
+                            style={currentViewedGroup === group ? {color: 'blue'} : {cursor: 'pointer'}}
+                            onClick={currentViewedGroup !== group ? async () => {await switchViewedGroup(group);} : undefined}
+                        >
+                            {group}
+                            {group !== TEST_GROUP && <Button onClick={async () => {await leaveGroup(group)}}>
```

Comment:
Don't need the async await on buttons generally speaking. Async/await is only if you need to do something after calling the awaited function